Automattic/newspack-blocks
# feat(donate): streamlined block option

### All Submissions:






### Changes proposed in this Pull Request:



Introduces an experimental, streamlined option for the Donate block. When activated, the donation flow will be as short as possible - the donor will only have to provide credit card details & email address. 

### TODOs

- [ ] Listen to Stripe successful payment via webhooks to send custom event to GA and update Campaigns data)

### How to test the changes in this Pull Request:

1. Set `newspack-plugin` to `feat/donation-block-streamlined` branch (https://github.com/Automattic/newspack-plugin/pull/1029)
2. Nothing should change yet, so test both donation platforms (WooCommerce (Newspack) and NRH) and confirm that donation flow works as before
3. Configure the site to use AMP Plus by adding the following env variable to `wp-config.php`: `define( 'NEWSPACK_AMP_PLUS_ENABLED', true );`
4. Set the Reader Revenue platform to NRH in the Reader Revenue wizard
5. Fill in the Stripe keys and choose a currency
6. On a page, insert a Donate block. Observe a new panel in editor sidebar - "Streamlined". Toggle the option on and publish the page
7. Observe that the Donate block features credit card & email inputs
8. [Test](https://stripe.com/docs/testing#cards) a successful transaction with a `4242 4242 4242 4242` card (and any valid other details) and observe a green "Thank you" notice and a successful charge in Stripe (and a Customer created)
9. Test a 3D Secure (requiring authentication) transaction with a `4000 0027 6000 3184` card
9. Test a failed transaction with a `4000000000000002` card

### Other information:
